Listener/CBLHTTPConnection.m

```diff
@@ -110,6 +112,26 @@ - (NSArray *)sslIdentityAndCertificates {
 }
 
 
+- (NSURL*) remoteURL {
+    if (!_builtRemoteURL) {
+        NSString* addr = asyncSocket.connectedHost;
+        if (![addr isEqualToString: @"127.0.0.1"]  && ![addr isEqualToString: @"::1"]) {
```

Comment:
I'm curious why we don't need to set source for localhost (for phonegap?)? If I run the listener on the simulator and request to the listener simulator by using real IP or hostname, I will get source but if I use 127.0.0.1, I will not get source.